processor/spanmetricsprocessor/factory.go

```diff
@@ -44,9 +46,10 @@ func createDefaultConfig() component.Config {
 		AggregationTemporality: "AGGREGATION_TEMPORALITY_CUMULATIVE",
 		DimensionsCacheSize:    defaultDimensionsCacheSize,
 		skipSanitizeLabel:      featuregate.GetRegistry().IsEnabled(dropSanitizationGateID),
+		MetricsFlushInterval:   15 * time.Second,
 	}
 }
 
 func createTracesProcessor(_ context.Context, params processor.CreateSettings, cfg component.Config, nextConsumer consumer.Traces) (processor.Traces, error) {
-	return newProcessor(params.Logger, cfg, nextConsumer)
+	return newProcessor(params.Logger, cfg, nextConsumer, clock.Realtime().NewTicker(cfg.(*Config).MetricsFlushInterval))
```

Comment:
To allow for using this within testing, you can use:

```suggestion
func createTracesProcessor(ctx context.Context, params processor.CreateSettings, cfg component.Config, nextConsumer consumer.Traces) (processor.Traces, error) {
	return newProcessor(params.Logger, cfg, nextConsumer, clock.FromContext(ctx).NewTicker(cfg.(*Config).MetricsFlushInterval))
```

Reply:
Good suggestion, done in: [75ddc2b](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17307/commits/75ddc2bfbbdd81f14716fa6b1adf9f62c7b06031)